src/main/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdk.java

```diff
@@ -0,0 +1,147 @@
+    this.managedSdkDirectory = managedSdkDirectory;
+    this.osInfo = osInfo;
+  }
+
+  public Path getSdkHome() {
+    return managedSdkDirectory.resolve(version.getVersion()).resolve("google-cloud-sdk");
+  }
+
+  /** Returns a path to gcloud executable (operating system specific). */
+  public Path getGcloud() {
+    return getSdkHome()
+        .resolve("bin")
+        .resolve(osInfo.name().equals(WINDOWS) ? "gcloud.cmd" : "gcloud");
+  }
+
+  /** Simple check to verify Cloud SDK installed by verifying the existence of gcloud. */
+  public boolean isInstalled() {
+    if (getSdkHome() == null) {
+      return false;
+    }
+    if (!Files.isDirectory(getSdkHome())) {
+      return false;
+    }
+    if (!Files.isRegularFile(getGcloud())) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Query gcloud to see if component is installed. Gcloud makes a call to the server to check this,
+   * in future version we can explore the use of '--local-state-only' but that's a relatively new
+   * flag and may not work for all cases.
+   */
+  public boolean hasComponent(SdkComponent component)
+      throws IOException, ExecutionException, CommandExitException {
+    if (!isInstalled()) {
+      return false;
+    }
+    List<String> listComponentCommand =
+        Arrays.asList(
+            getGcloud().toString(),
+            "components",
+            "list",
+            "--format=json",
+            "--filter=id:" + component + " AND state.name:Not Installed");
+
+    String result = new CommandFactory().newCaller(listComponentCommand, null, null).call();
+    return !result.contains(component.toString());
+  }
+
+  /** Query gcloud to see if sdk is up to date. Gcloud makes a call to the server to check this. */
+  public boolean isUpToDate() throws IOException, ExecutionException, CommandExitException {
+    if (!isInstalled()) {
+      return false;
+    }
+    if (version != Version.LATEST) {
+      return true;
+    }
+
+    List<String> updateAvailableCommand =
+        Arrays.asList(
+            getGcloud().toString(),
+            "components",
+            "list",
+            "--format=json",
+            "--filter=state.name:Update Available");
+
+    String result = new CommandFactory().newCaller(updateAvailableCommand, null, null).call();
+    return !result.contains("Update Available");
+  }
+
+  // TODO : fix passthrough for useragent and client side usage reporting
+  public SdkInstaller newInstaller() throws UnsupportedOsException {
+    String userAgentString = "google-cloud-tools-java";
+    return SdkInstaller.newInstaller(managedSdkDirectory, version, osInfo, userAgentString, false);
+  }
+
+  public SdkComponentInstaller newComponentInstaller() {
+    return SdkComponentInstaller.newComponentInstaller(getGcloud());
+  }
+
+  /** Return a new updater if a 'LATEST' sdk, returns {@code null} for versioned SDKs. */
+  public SdkUpdater newUpdater() {
+    if (version != Version.LATEST) {
+      return null;
```

Comment:
Returning null here feels iffy. I'd expect this method to return an object unconditionally and a separate check be made at a higher level to decide whether this method call is needed.